src/components/Request.tsx

```diff
@@ -1,63 +1,82 @@
 import React from "react";
-import { handleForm, getFormValue } from "../utils";
-import { Message, Network } from "many";
-import { CoseMessage } from "many/dist/message/cose";
+import {
+  Ed25519KeyPairIdentity as Id,
+  Message as Msg,
+  Network,
+} from "@liftedinit/many-js";
+import {
+  Box,
+  Button,
+  Heading,
+  Tab,
+  TabList,
+  TabPanel,
+  TabPanels,
+  Tabs,
+  Textarea,
+} from "@liftedinit/ui";
 
-import Section from "./Section";
-import ButtonGroup from "./ButtonGroup";
-import Button from "./Button";
-import Tabs from "./Tabs";
-import Tab from "./Tab";
-
-const sendRequest = (url: string) => async (form: HTMLFormElement) => {
-  const hex = getFormValue(form, "hex");
+const sendReq = async (url: string, hex: string) => {
   const network = new Network(url);
   const res = await network.sendEncoded(Buffer.from(hex, "hex"));
   return res.toString("hex");
 };
 
 interface RequestProps {
-  req: string;
-  setRes: (res: string) => void;
   url: string;
+  id: Id | undefined;
+  msg?: Msg;
+  setRes: (res: string) => void;
 }
 
-function Request({ req, setRes, url }: RequestProps) {
-  const [activeTab, setActiveTab] = React.useState(0);
-  const cose = req.length
-    ? CoseMessage.fromCborData(Buffer.from(req, "hex")).toString()
-    : "";
-  const message = req.length
-    ? Message.fromCborData(Buffer.from(req, "hex")).toString()
-    : "";
+function Request({ url, id, msg, setRes }: RequestProps) {
+  const [hex, setHex] = React.useState("");
+  const [json, setJson] = React.useState("");
+
+  React.useEffect(() => {
+    async function convertMsg(msg: Msg) {
+      const cose = await msg?.toCoseMessage(id);
+      setHex(cose.toCborData().toString("hex"));
+    }
+    msg && convertMsg(msg);
+  }, [id, msg]);
+
+  React.useEffect(() => {
+    function hexToJson(hex: string) {
+      const cose = Msg.fromCborData(Buffer.from(hex, "hex"));
+      setJson(cose.toString());
+    }
+    hex && hexToJson(hex);
+  }, [hex]);
```

Comment:
You can generate a message from the form _or_ paste hex in directly. Either way will be parsed into the JSON view and set the pending request.